src/lib/operators/join_mpsm/column_materializer_numa.hpp

```diff
@@ -0,0 +1,202 @@
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "resolve_type.hpp"
+#include "scheduler/current_scheduler.hpp"
+#include "scheduler/job_task.hpp"
+#include "scheduler/topology.hpp"
+#include "storage/create_iterable_from_column.hpp"
+#if HYRISE_NUMA_SUPPORT
+#include "storage/numa_placement_manager.hpp"
+#else
+#include "utils/boost_default_memory_resource.cpp"
+#endif
+#include "types.hpp"
+#include "utils/numa_memory_resource.hpp"
+
+namespace opossum {
+
+template <typename T>
+struct MaterializedValue {
+  MaterializedValue() = default;
+  MaterializedValue(RowID row, T v) : row_id{row}, value{v} {}
+
+  RowID row_id;
+  T value;
+};
+
+template <typename T>
+using MaterializedValueAllocator = PolymorphicAllocator<MaterializedValue<T>>;
+
+template <typename T>
+using MaterializedChunk = std::vector<MaterializedValue<T>, MaterializedValueAllocator<T>>;
+
+template <typename T>
+struct MaterializedNUMAPartition {
+  NodeID _node_id;
+  MaterializedValueAllocator<T> _alloc;
+  std::vector<std::shared_ptr<MaterializedChunk<T>>> _chunk_columns;
+
+  explicit MaterializedNUMAPartition(NodeID node_id, size_t reserve_size)
+      : _node_id{node_id},
+#if HYRISE_NUMA_SUPPORT
+        _alloc{NUMAPlacementManager::get().get_memory_resource(node_id)},
```

Comment:
Shouldn't we move this somewhere else? As more and more operators become NUMA-aware, I can see this plopping up everywhere.